### PR TITLE
kubeadm: simply struct of `joinOptions` by removing `joinControlPlane`

### PR DESCRIPTION
**Pull request by chendave**
Signed-off-by: Dave Chen <dave.chen@arm.com>

#### What type of PR is this?

/kind cleanup

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: cmd/kubeadm/app/cmd/join_test.go [hunk 1]
```diff
@@ -17,13 +17,15 @@ limitations under the License.
...
+	"bytes"
 	"fmt"
 	"os"
 	"path/filepath"
+	"strings"
 	"testing"
 
 	"k8s.io/apimachinery/pkg/util/sets"
-
+	"k8s.io/klog/v2"
```
Comment (neolit123) [inline cmd/kubeadm/app/cmd/join_test.go:28]: should be under the apimachinery line above and have empty line below it.

File: cmd/kubeadm/app/cmd/join_test.go [hunk 2]
```diff
@@ -246,20 +252,53 @@ func TestNewJoinData(t *testing.T) {
...
 				cmd.Flags().Set(f, v)
 			}
 
 			// test newJoinData method
 			data, err := newJoinData(cmd, tc.args, joinOptions, nil, kubeconfigFilePath)
+			klog.Flush()
+			msg := "WARNING: --control-plane is also required when passing control-plane"
+			if tc.expectWarn {
+				if !strings.Contains(buffer.String(), msg) {
+					t.Errorf("Haven't detected the warning message, expected: %v, acutal: %v", msg, buffer.String())
```
Comment (neolit123) [inline cmd/kubeadm/app/cmd/join_test.go:295]: typo in actual

**Review by neolit123 — COMMENTED** [review]
so this SGTM but would expect another reviewer to double check. hopefully we are not breaking anything.

/approve

**Review by neolit123 — COMMENTED** [review]
/hold

File: cmd/kubeadm/app/cmd/join_test.go (hunk 1 above)
Comment (chendave) [reply cmd/kubeadm/app/cmd/join_test.go:28]: Done

File: cmd/kubeadm/app/cmd/join_test.go (hunk 2 above)
Comment (chendave) [reply cmd/kubeadm/app/cmd/join_test.go]: Done

**Review by pacoxu — APPROVED** [review]
/lgtm

`cfg.ControlPlane.CertificateKey,` is similar to `cfg.NodeRegistration.Name` that does not needs another joinControlPlane in joinOptions.

**Comment by chendave** [comment]
assume it's fine to lift the hold label now. :)

/hold cancel

**Comment by chendave** [comment]
irrelevant flaky

/test pull-kubernetes-integration

**Pull request merged**